addons/crm_iap_lead/models/crm_lead.py

```diff
@@ -1,11 +1,141 @@
+        response_clearbit = record._make_request(request)
+        self._enrich_leads_from_response(response_clearbit)
+
+    @api.multi
+    def lead_enrich_mail_server_action(self):
+        # This function has as purpose to differentiate the way the errors are handeled if the code is called by
+        # A server action or using the normal flow. This beacuse InsufficientCreditError are not handeled in the server actions.
+        try:
+            return self.lead_enrich_mail()
+        except InsufficientCreditError:
+            raise UserError("You do not have enough credits to perform this action.")  
+
+    def _make_request(self, domains):
+        """This methode will query the endpoint to get the data for the asked (lead.id, domain) pairs"""
+        reveal_account = self.env['iap.account'].get('reveal')
+        dbuuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('reveal.endpoint', DEFAULT_ENDPOINT) + '/iap/clearbit/1/lead_enrichment_email'
+        params = {
+            'account_token': reveal_account.account_token,
+            'dbuuid': dbuuid,
+            'domains': domains,
+        }
+        return jsonrpc(endpoint, params=params, timeout=300)
+
+    def _create_message_data(self, company_data):
+        log_data = {
+            'message_title': "Lead enriched based on email",
+            'twitter': company_data['twitter'],
+            'description': company_data['description'],
+            'logo': company_data['logo'],
+            'name': company_data['name'],
+            'phone_numbers': company_data['phone_numbers'],
+            'facebook': company_data['facebook'],
+            'linkedin': company_data['linkedin'],
+            'crunchbase': company_data['crunchbase'],
+            'tech': [t.replace('_', ' ').title() for t in company_data['tech']],
+        }
+        timezone = company_data['timezone']
+        if timezone:
+            log_data.update({
+                'timezone': timezone.replace('_', ' ').title(),
+                'timezone_url': company_data['timezone_url'],
+            })
+        return log_data
+
+    def _send_message(self, company_data = False):
+        if company_data:
+            messages_to_post = self._create_message_data(company_data)
+            self.message_post_with_view('crm_iap_lead.lead_message_template', values=messages_to_post, subtype_id=self.env.ref('mail.mt_note').id)
+        else:
+            self.message_post_with_view('crm_iap_lead.lead_message_not_found', subtype_id=self.env.ref('mail.mt_note').id)
+
+    @api.multi
+    def _enrich_leads_from_response(self, data_clearbit):
+        """ This method will get the response from the service and enrich the lead accordingly """
+        for lead_id, data in data_clearbit.items():
+            record = self.browse(int(lead_id))
+            if record.exists():
+                if data:
+                    street = False
```

Comment:
useless variable, in your dict, do
`'street': record.street or data['street_name']`